packages/rocketchat-file/file.server.js

```diff
@@ -0,0 +1,300 @@
+	return {
+		image: imageData[1],
+		contentType: imageData[0].replace('data:', '')
+	};
+};
+
+RocketChatFile.addPassThrough = function(st, fn) {
+	const pass = new stream.PassThrough();
+	fn(pass, st);
+	return pass;
+};
+
+RocketChatFile.GridFS = class {
+	constructor(config) {
+		if (config == null) {
+			config = {};
+		}
+		let name = config.name;
+		const transformWrite = config.transformWrite;
+		if (name == null) {
+			name = 'file';
+		}
+		this.name = name;
+		this.transformWrite = transformWrite;
+		const mongo = Package.mongo.MongoInternals.NpmModule;
+		const db = Package.mongo.MongoInternals.defaultRemoteCollectionDriver().mongo.db;
+		this.store = new Grid(db, mongo);
+		this.findOneSync = Meteor.wrapAsync(this.store.collection(this.name).findOne.bind(this.store.collection(this.name)));
+		this.removeSync = Meteor.wrapAsync(this.store.remove.bind(this.store));
+		this.getFileSync = Meteor.wrapAsync(this.getFile.bind(this));
+	}
+
+	findOne(fileName) {
+		return this.findOneSync({
+			_id: fileName
+		});
+	}
+
+	remove(fileName) {
+		return this.removeSync({
+			_id: fileName,
+			root: this.name
+		});
+	}
+
+	createWriteStream(fileName, contentType) {
+		const self = this;
+		let ws = this.store.createWriteStream({
+			_id: fileName,
+			filename: fileName,
+			mode: 'w',
+			root: this.name,
+			content_type: contentType
+		});
+		if (self.transformWrite != null) {
+			ws = RocketChatFile.addPassThrough(ws, function(rs, ws) {
+				const file = {
+					name: self.name,
+					fileName,
+					contentType
+				};
+				return self.transformWrite(file, rs, ws);
+			});
+		}
+		ws.on('close', function() {
+			return ws.emit('end');
+		});
+		return ws;
+	}
+
+	createReadStream(fileName) {
+		return this.store.createReadStream({
+			_id: fileName,
+			root: this.name
+		});
+	}
+
+	getFileWithReadStream(fileName) {
+		const file = this.findOne(fileName);
+		if (file == null) {
+			return null;
+		}
+		const rs = this.createReadStream(fileName);
+		return {
+			readStream: rs,
+			contentType: file.contentType,
+			length: file.length,
+			uploadDate: file.uploadDate
+		};
+	}
+
+	getFile(fileName, cb) {
+		const file = this.getFileWithReadStream(fileName);
+		if (!file) {
+			return cb();
+		}
+		const data = [];
+		file.readStream.on('data', Meteor.bindEnvironment(function(chunk) {
+			return data.push(chunk);
+		}));
+		return file.readStream.on('end', Meteor.bindEnvironment(function() {
+			return cb(null, {
+				buffer: Buffer.concat(data),
+				contentType: file.contentType,
+				length: file.length,
+				uploadDate: file.uploadDate
+			});
+		}));
+	}
+
+	deleteFile(fileName) {
+		const file = this.findOne(fileName);
+		if (file == null) {
+			return null;
+		}
+		return this.remove(fileName);
+	}
+
+
+};
+
+RocketChatFile.FileSystem = class {
+	constructor(config) {
+		if (config == null) {
+			config = {};
+		}
+		let absolutePath = config.absolutePath;
+		const transformWrite = config.transformWrite;
```

Comment:
You can use const `const {absolutePath = '~/uploads', transformWrite} = config`;